yt/frontends/amrvac/tests/test_outputs_pytest.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np  # NOQA
+import pytest
+
+import yt  # NOQA
+from yt.frontends.amrvac.api import AMRVACDataset, AMRVACGrid
+from yt.testing import requires_file
+from yt.units import YTArray
+from yt.utilities.answer_testing import utils
+from yt.utilities.answer_testing.answer_tests import (
+    field_values,
+    grid_hierarchy,
+    grid_values,
+    parentage_relationships,
+    projection_values,
+)
+
+# Test parameters
+blastwave_spherical_2D = "amrvac/bw_2d0000.dat"
+khi_cartesian_2D = "amrvac/kh_2d0000.dat"
+khi_cartesian_3D = "amrvac/kh_3D0000.dat"
+jet_cylindrical_25D = "amrvac/Jet0003.dat"
+riemann_cartesian_175D = "amrvac/R_1d0005.dat"
+blastwave_cartesian_3D = "amrvac/bw_3d0000.dat"
+blastwave_polar_2D = "amrvac/bw_polar_2D0000.dat"
+blastwave_cylindrical_3D = "amrvac/bw_cylindrical_3D0000.dat"
+rmi_cartesian_dust_2D = "amrvac/Richtmyer_Meshkov_dust_2D/RM2D_dust_Kwok0000.dat"
+
+
+ds_list = [
+    blastwave_spherical_2D,
+    khi_cartesian_2D,
+    khi_cartesian_3D,
+    jet_cylindrical_25D,
+    riemann_cartesian_175D,
+    blastwave_cartesian_3D,
+    blastwave_polar_2D,
+    blastwave_cylindrical_3D,
+    rmi_cartesian_dust_2D,
+]
+a_list = [0, 1, 2]
+w_list = [None, "density"]
+dso_list = [None, ("sphere", ("max", (0.1, "unitary")))]
+
+
+def _get_fields_to_check(fname):
+    # This function is called during test collection. If this frontend
+    # is not being run, and therefore the data isn't present, this try
+    # except block prevents pytest from failing needlessly
+    try:
+        ds = utils.data_dir_load(fname)
+        fields = [("gas", "density"), ("gas", "velocity_magnitude")]
+        field_ids = ["density", "velocity_magnitude"]
+        raw_fields_labels = [fname for ftype, fname in ds.field_list]
+        if "b1" in raw_fields_labels:
+            fields.append(("gas", "magnetic_energy_density"))
+            field_ids.append("magnetic_energy_density")
+        if "e" in raw_fields_labels:
+            fields.append(("gas", "energy_density"))
+            field_ids.append("energy_density")
+        if "rhod1" in raw_fields_labels:
+            fields.append(("gas", "total_dust_density"))
+            field_ids.append("total_dust_density")
+            # note : not hitting dust velocity fields
+        # return [fields, field_ids]
+        return [ds, fields]
+    except FileNotFoundError:
+        return [None, [None,]]
```

Comment:
This should rather be moved upward, as close as possible from `ds = utils.data_dir_load(fname)`.